api/src/main/java/org/pmiops/workbench/opsgenie/EgressEventServiceImpl.java

```diff
@@ -52,22 +55,36 @@ public EgressEventServiceImpl(
       Provider<AlertApi> alertApiProvider,
       Provider<WorkbenchConfig> workbenchConfigProvider,
       UserService userService,
-      WorkspaceAdminService workspaceAdminService) {
+      WorkspaceAdminService workspaceAdminService,
+      WorkspaceDao workspaceDao) {
     this.clock = clock;
     this.egressEventAuditor = egressEventAuditor;
     this.institutionService = institutionService;
     this.alertApiProvider = alertApiProvider;
     this.workbenchConfigProvider = workbenchConfigProvider;
     this.userService = userService;
     this.workspaceAdminService = workspaceAdminService;
+    this.workspaceDao = workspaceDao;
   }
 
   @Override
   public void handleEvent(EgressEvent event) {
+    // Lookup workspace by googleProject name, and set the workspaceNamespace in EgressEvent.
+    String workspaceNamespace =
+        workspaceDao
+            .getByGoogleProject(event.getProjectName())
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
```

Comment:
I don't want the egress alert to fail and be suppressed in this scenario. There might be some abusive scenarios where our DB shows a deleted workspace, but egress is still being generated - i.e. the project and/or VMs still exist somehow. If we throw on this line, we don't see any error.

Please just log the fact that it wasn't found (this is still unexpected) and then leave the namespace as null, "", or some sentinel value to indicate it wasn't found.